lib/topological_inventory/ansible_tower/targeted_refresh/service_instance.rb

```diff
@@ -0,0 +1,202 @@
+require "topological_inventory/ansible_tower/collector"
+require "topological_inventory/providers/common/operations/sources_api_client"
+require "topological_inventory/ansible_tower/receptor/async_receiver"
+
+module TopologicalInventory
+  module AnsibleTower
+    module TargetedRefresh
+      class ServiceInstance < TopologicalInventory::AnsibleTower::Collector
+        REFS_PER_REQUEST_LIMIT = 20
+
+        def initialize(payload = {})
+          self.params    = payload['params']
+          self.source_id = payload['source_id']
+
+          # TODO: add metrics exporter
+          super(payload['source_uid'], nil)
+        end
+
+        # Entrypoint for 'ServiceInstance.refresh' operation
+        def refresh
+          return if params_missing?
+
+          # Connection settings
+          set_connection_data!
+          logger.info("ServiceInstance#refresh - Connection set successfully")
+
+          # Get all tasks
+          tasks, refresh_state_uuid = {}, SecureRandom.uuid
+          params.to_a.each do |task|
+            if task['task_id'].blank? || task['source_ref'].blank?
+              logger.warn("ServiceInstance#refresh - missing data for task: #{task}")
+              next
+            end
+            tasks[task['task_id']] = task['source_ref']
+
+            if tasks.length == REFS_PER_REQUEST_LIMIT
```

Comment:
Is this tied to a paging limit of some type? Not sure why the number 20 is the default.

Reply:
@syncrou I just specify something. The only limitation is max length of URL (which can't be computed in on-premise mode).
I guess 100 could be also fine. What do you suggest?

Reply:
20 is probably fine. I was more interested in now that came to be. Seems like it can't be computed so any number as a default is okay.